src/uu/pgrep/src/process_matcher.rs

```diff
@@ -76,6 +80,26 @@ pub fn get_match_settings(matches: &ArgMatches) -> UResult<Settings> {
         gid: matches
             .get_many::<u32>("group")
             .map(|ids| ids.cloned().collect()),
+        pgroup: matches.get_many::<u64>("pgroup").map(|xs| {
+            xs.map(|pg| {
+                if *pg == 0 {
+                    unsafe { getpgrp() as u64 }
+                } else {
```

Comment:
Could you add some safety explain here? :)

see also: https://github.com/uutils/coreutils/blob/main/CONTRIBUTING.md#unsafe